### DISCUSSION TITLE
Process for keeping docker/docker-compose.yml updated

### DISCUSSION THREAD

**Discussion by sabman** [supabase/supabase#17074] [General]
# Process for keeping `docker/docker-compose.yml` updated

## Describe the chore

This https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml is out of date with the latest releases of most images. We need to define a process that keeps this up-to-date with release versions. Additional challenge is to ensure these versions are compatible with each other. Another thing I noticed is that studio isn't strictly following a release tag. It seems features are deployed as they are merged into master.

## Proposed solutions

1. Manually update and create a PR on a regular bases
2. Automate based on github actions triggering an update to the docker-compose file and creating a PR

Any other ideas?

**Comment by kiwicopple** [comment]
would love to hear any ideas for automations. We could always use a `latest` tag, but it's very secure.

I can also just add some docs to suggest updating the docker-compose file before running a `docker pull`

**Reply by sabman** [reply]
We could make docker compose file a template file where the tags come from a dot env file. Then we keep the dot env file up to date. In our release notes we have instructions to update this file. I would start with this and then iterate to automate this. Small bites ☺️🍱

BTW what do you mean latest is secure? Is it not public?

**Comment by sabman** [comment]
@kiwicopple did we end up doing this? Also what is source of truth to know the compatible versions for use in `docker-compose.yml`?
> I can also just add some docs to suggest updating the docker-compose file before running a `docker pull`
